smartcontractkit/external-adapters-js
# Add local cache fallback to websockets

## Description

PR #1278 added a fallback to local cache for requests when the Redis cache threw an error. Due to the way the WebSocket framework is wired up, those changes applied to the background websocket messages received (i.e. they were correctly set to the local cache in case of redis errors) but the base request to a WS adapter still failed to use this fallback and returned a 500 to the client.

This PR copies that fallback logic, taking advantage of the singleton nature of the local cache implementation.

......

## Changes

- Add local cache fallback to WS request resolution

## Steps to Test

1. Start a WS only adapter (e.g. ncfx) with Redis cache enabled
2. Make a request to it, verifying that the Redis connection is kept alive
3. Stop/terminate the Redis instance
4. Background pushes from the server should continue to be cached in local cache
5. Requests to that adapter should no longer fail surfacing the Redis error, but succeeding instead

## Quality Assurance
